tools/validate_spec.py

```diff
@@ -122,10 +146,13 @@ def validate(
             directory = sorted(directory_path.glob("*")) if directory_path is not None else []
             for path in directory:
                 validation_result = _validate_message(path, is_valid_directory, message_value)
+                if coverage_info is not None:
+                    coverage_info.update(validation_result.parser_result)
                 output_writer.write_result(validation_result)
                 if not validation_result.validation_success and abort_on_error:
                     raise ValidationError(f"aborted: message {path} was classified incorrectly")
-
+        if coverage_info is not None:
+            output_writer.write_coverage(coverage_info)
```

Comment:
The `is not None` checks could be done inside the method. The readability just gets worse with these checks.

Reply:
Done in ad03d50